pkg/server/skydb/pq/builder/factory.go

```diff
@@ -326,3 +332,34 @@ type joinedTable struct {
 func (a joinedTable) equal(b joinedTable) bool {
 	return a.secondaryTable == b.secondaryTable && a.primaryColumn == b.primaryColumn && a.secondaryColumn == b.secondaryColumn
 }
+
+func (f *predicateSqlizerFactory) NewSort(s skydb.Sort) (string, error) {
+	expr := s.Expression
+	if expr.IsKeyPath() && len(expr.KeyPathComponents()) == 2 {
```

Comment:
Please handle `len(expr.KeyPathComponents()) > 2` explicitly.

Reply:
Updated to not support expr.KeyPathComponents() > 2 for now same as the current predicate expression.